### PR TITLE
Notifications

### PR DESCRIPTION
**Pull request by dshanske**
This is the initial creation of support for improving comment email notifications for webmentions. As of now, it only displays a customized display for webmentions that is very similar to the default comment display.

As we add in more rendering over PHP-MF2, it will expand to show more. 

The version in Semantic Linkbacks uses the older filter, whereas this uses the newer hook to take control and either pass it to the core function, or a new local implementation based on comment type.

### CODE REVIEW

**Comment by dshanske** [comment]
@pfefferle Any thoughts on this?

**Comment by pfefferle** [comment]
Why do we have to copy nearly the whole functionality?

**Comment by dshanske** [comment]
Because there is no easy way otherwise. I am thinking of submitting a core patch

**Comment by pfefferle** [comment]
Why do the "normal" notification not work?

**Comment by dshanske** [comment]
Because it doesn't display the relevant fields or handle custom comment types

File: includes/class-webmention-notifications.php [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+			@wp_mail( $email, wp_specialchars_decode( $subject ), $notify_message, $message_headers );
+		}
+
+		if ( $switched_locale ) {
+			restore_previous_locale();
+		}
+		return true;
+	}
+
+
```
Comment (pfefferle) [inline includes/class-webmention-notifications.php]: can you add the missing docblock?

File: includes/class-webmention-notifications.php [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+				$message .= sprintf( __( 'URL: %s', 'webmention' ), $comment->comment_author_url ) . "\r\n";
+				/* translators: %s: comment text */
+				$message .= sprintf( __( 'Comment: %s', 'webmention' ), "\r\n" . get_comment_text( $comment ) ) . "\r\n\r\n";
+				break;
+		}
+		return apply_filters( 'webmention_comment_details', $message, $comment );
+
+	}
+
+
```
Comment (pfefferle) [inline includes/class-webmention-notifications.php]: Docblock

File: includes/class-webmention-notifications.php [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+					break;
+				case 'resend_moderation_email':
+					self::wp_notify_moderator( $comment_id );
+					break;
+			}
+		}
+		return $redirect_to;
+	}
+
+	/*
```
Comment (pfefferle) [inline includes/class-webmention-notifications.php]: Docblocks start with /**

File: includes/class-webmention-notifications.php [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+
+	}
+
+
+	public static function wp_new_comment_notify_postauthor( $comment ) {
+		$comment = get_comment( $comment );
+
+		$maybe_notify = get_option( 'comments_notify' );
+
+		/**
```
Comment (pfefferle) [inline includes/class-webmention-notifications.php]: I am not sure, but isn't WordPress using `/*` for inline multiline docs?

**Pull request merged**